modules/programs/kodi.nix

```
@@ -0,0 +1,238 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.kodi;
+
+  stylesheetCommonHeader = ''
+    <?xml version="1.0"?>
+    <xsl:stylesheet xmlns:xsl='http://www.w3.org/1999/XSL/Transform' version='1.0'>
+    <xsl:output method="xml" indent="yes" omit-xml-declaration="yes"/>
+    <xsl:strip-space elements="*"/>
+  '';
+
+  stylesheetCommonFooter = "</xsl:stylesheet>";
+
+  stylesheetNestedTags = ''
+    <xsl:template match="attr[attrs]">
+      <xsl:variable name="elementName" select="@name"/>
+      <xsl:element name="{$elementName}">
+        <xsl:apply-templates select="attrs" />
+      </xsl:element>
+    </xsl:template>
+    <xsl:template match="attr[list[attrs]]">
+    <xsl:variable name="elementName" select="@name"/>
+    <xsl:for-each select="list/attrs">
+      <xsl:element name="{$elementName}">
+        <xsl:apply-templates select="." />
+      </xsl:element>
+    </xsl:for-each>
+    </xsl:template>
+    <xsl:template match="attr[not(attrs|list)]">
+      <xsl:variable name="elementName" select="@name"/>
+      <xsl:element name="{$elementName}">
+      <xsl:if test="$elementName='path'">
+        <!-- needed in sources.xml but will be used for all "path" tags -->
+        <xsl:attribute name="pathversion">1</xsl:attribute>
+      </xsl:if>
+        <xsl:value-of select="*/@value" />
+      </xsl:element>
+    </xsl:template>
+  '';
+
+  stylesheetTagsAsSettingWithId = ''
+    <xsl:template match='attr'>
+      <setting>
+        <xsl:attribute name="id">
+          <xsl:value-of select="@name" />
+        </xsl:attribute>
+        <xsl:value-of select="*/@value" />
+      </setting>
+    </xsl:template>
+  '';
+
+  stylesheetAdvancedSettingsRootTag = ''
+    <xsl:template match='/'>
+      <xsl:comment> Generated by Home Manager. </xsl:comment>
+      <advancedsettings version="1.0">
+        <xsl:apply-templates/>
+      </advancedsettings>
+    </xsl:template>
+  '';
+
+  stylesheetSourcesRootTag = ''
+    <xsl:template match='/'>
+      <xsl:comment> Generated by Home Manager. </xsl:comment>
+      <sources>
+        <xsl:apply-templates/>
+      </sources>
+    </xsl:template>
+  '';
+
+  stylesheetAddonSettingsRootTag = ''
+    <xsl:template match='/'>
+      <xsl:comment> Generated by Home Manager. </xsl:comment>
+      <settings version="2">
+        <xsl:apply-templates/>
+      </settings>
+    </xsl:template>
+  '';
+
+  attrsetToXml = attrs: name: stylesheet:
+    pkgs.stdenv.mkDerivation {
+      inherit name stylesheet;
+      nativeBuildInputs = [ pkgs.libxslt ];
+      phases = [ "installPhase" ];
+      installPhase = let xml = (builtins.toXML attrs);
+      in ''
+        xsltproc ${stylesheet} - <<EOF > $out
+        ${xml}
+        EOF
+      '';
```

I think it probably is a bit clearer to simply inline the `xml` variable:
```suggestion
      installPhase = ''
        xsltproc ${stylesheet} - <<EOF > $out
        ${builtins.toXML attrs}
        EOF
      '';
```